### PR TITLE
snap/pack: fix pack with integrity unit test on openSUSE

### PR DESCRIPTION
**Pull request by bboozzoo**
Apparently on openSUSE, the squashfs image file is padded to 64k by default, see https://build.opensuse.org/package/view_file/filesystems/squashfs/squashfs-64k.patch?expand=1 

This fixes the unit test failure on openSUSE blocking the package update since 2.60.

### CODE REVIEW

File: snap/pack/pack_test.go [hunk 1]
```diff
@@ -588,15 +588,24 @@ esac
...
+		integrityStartOffset = 65536
+	}
+
 	// example snap has a size of 16384 (4 blocks)
-	_, err = snapFile.Seek(squashfs.MinimumSnapSize, io.SeekStart)
+	_, err = snapFile.Seek(integrityStartOffset, io.SeekStart)
 	c.Assert(err, IsNil)
 
 	integrityHdr := make([]byte, integrity.HeaderSize)
 	_, err = snapFile.Read(integrityHdr)
```
Comment (zyga) [inline snap/pack/pack_test.go:605]: Looks good

**Review by zyga — APPROVED** [review]

**Comment by xnox** [comment]
Oh, maybe Debian and Ubuntu need a fix here. We do get warnings that base:bare is too small, I didn't know one can pad the squashfs which may make those warnings go away.

**Comment by bboozzoo** [comment]
I may have been wrong as to the default padding thing on various distros. Looks like opensuse has a patch to enable padding up to 64k in their tree. I'll update the patch and the commit message to reflect that.

**Comment by bboozzoo** [comment]
I've updated this and #13376

**Comment by xnox** [comment]
> I may have been wrong as to the default padding thing on various distros. Looks like opensuse has a patch to enable padding up to 64k in their tree. I'll update the patch and the commit message to reflect that.

Even then it seems like the right thing to do! I'll browse OBS and try to see if we can integrate that patch into Ubuntu too, or like submit it upstream too such that we are all on the same page (pun intended ;-) )

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**